### PR TITLE
feat: Add support for git hash in the app and app group overview section 

### PR DESCRIPTION
**Pull request by RajeevRanjan27**
# Description
This include feature of getting the githash of artifacts deployed fot that app in app overview environment section
Also the same feature is there in the app group overview section too as requested.
 

Fixes devtron-labs/devtron#4797

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/app/AppListingService.go [hunk 1]
```diff
@@ -99,6 +99,10 @@ type AppNameTypeIdContainer struct {
 	Type    string `json:"type"`
 	AppId   int    `json:"appId"`
 }
+type CiArtifactWithParentArtifact struct {
```
Comment (vikramdevtron) [inline pkg/app/AppListingService.go:102]: This should move at some common place

File: pkg/app/AppListingService.go [hunk 2]
```diff
@@ -769,6 +802,91 @@ func (impl AppListingServiceImpl) FetchAppStageStatus(appId int, appType int) ([
...
+		} else {
+			artifactChildParentMap[artifact.CiArtifactId] = artifact.CiArtifactId
+		}
+
+		// Ensure uniqueness of artifact IDs in the slice.
+		if ciArtifactIds == nil {
+			if _, ok := artifactChildParentMap[artifact.CiArtifactId]; ok {
+				ciArtifactIds = []int{artifactChildParentMap[artifact.CiArtifactId]}
+			}
+		} else if !slices.Contains(ciArtifactIds, artifactChildParentMap[artifact.CiArtifactId]) {
```
Comment (nishant-d) [inline pkg/app/AppListingService.go:849]: `Contains` is a linear search, invoking inside `for` loop will  make complexity exponential.

File: pkg/app/AppListingService.go [hunk 3]
```diff
@@ -806,6 +941,15 @@ func (impl AppListingServiceImpl) FetchOtherEnvironment(ctx context.Context, app
 		} else {
 			env.ChartRefId = chart.ChartRefId
 		}
+
+		gitCommits, exists := gitCommitsWithArtifacts[env.CiArtifactId]
+		if exists {
```
Comment (nishant-d) [inline pkg/app/AppListingService.go:946]: rewrite it for only one assignment and one initialisation

File: specs/global-plugin-system.yaml [hunk 4]
```diff
@@ -0,0 +1,615 @@
+openapi: "3.0.3"
```
Comment (nishant-d) [inline specs/global-plugin-system.yaml:1]: is it related ?

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: specs/global-plugin-system.yaml (hunk 4 above)
Comment (RajeevRanjan27) [reply specs/global-plugin-system.yaml:1]: not related

File: pkg/app/AppListingService.go (hunk 3 above)
Comment (RajeevRanjan27) [reply pkg/app/AppListingService.go]: solved

File: pkg/app/AppListingService.go (hunk 2 above)
Comment (RajeevRanjan27) [reply pkg/app/AppListingService.go]: resolved using map

**Pull request closed without merging**